tests/integ/test_stored_procedure.py

```diff
@@ -239,6 +239,20 @@ def sp_pow(session_, x, y):
     assert pow_sp(2, 10, session=session) == 1024
 
 
+def test_basic_snowpark_pandas_stored_procedure(session):
+    def return1(session_):
```

Comment:
I thought the goal was to test that `session` is not required.

Reply:
Before this PR, the Snowpark python API would still require having a session argument for the sproc's handler. I updated the PR to allow for optional session argument even when using the Snowpark python API and added tests for both cases (with and without the session argument).

@sfc-gh-sfan would be great if you can confirm that the update to Snowpark python does make sense.